elasticdl/python/worker/worker.py

```diff
@@ -308,27 +311,32 @@ def get_model(self):
             if not res.model_init_status:
                 # push variable to ps for initialization
                 self.report_variable_to_ps(ps_id)
+                req = elasticdl_pb2.PullVariableRequest()
+                req.current_model_version = self._model_versions_from_ps[ps_id]
                 res = self._ps_stubs[ps_id].pull_variable(req)
                 if not res.model_init_status:
                     # TODO: support PS fault-tolerance
                     raise RuntimeError(
                         "PS pod %d cannot be initialized" % ps_id
                     )
 
-            for tensor_pb in res.model.param:
-                tensor = Tensor.from_tensor_pb(tensor_pb)
-                self._non_embed_vars[tensor.name].assign(tensor.to_ndarray())
+            if res.model.version > self._model_versions_from_ps[ps_id]:
```

Comment:
If `res.model_version <= self._model_versions_from_ps[ps_id]`, the parameter server will not set mdel.param. So this field is empty. The for loop will be skipped. It seems that the logic here in worker is duplicate with PS.

Reply:
You are right. No need to check version here.